### PR TITLE
ICDS: Update to Beneficiary Indicator #2

### PR DESCRIPTION
**Pull request by gcapalbo**
- allows for custom content handlers in new framework
- makes second beneficiary indicator use a custom content handler to inspect latest growth monitoring form for content
- allows switching schedules to be live instead of always being in test mode
- starts capturing the icds indicator slug along with each SMS sent

@emord

### CODE REVIEW

File: custom/icds/messaging/custom_content.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    forms_under_consideration = []
+
+    db_names = get_db_aliases_for_partitioned_query()
+    for db_name in db_names:
+        result = XFormInstanceSQL.objects.using(db_name).filter(
+            domain=domain,
+            form_id__in=form_ids,
+            xmlns=GROWTH_MONITORING_XMLNS,
+            state=XFormInstanceSQL.NORMAL,
+        ).order_by('-received_on')[0:1]
```
Comment (emord) [inline custom/icds/messaging/custom_content.py]: Could you use `.first()`  here?

File: corehq/apps/sms/api.py [hunk 2]
```diff
@@ -55,17 +55,17 @@ def __init__(self, *args, **kwargs):
...
 def add_msg_tags(msg, metadata):
     if msg and metadata:
         fields = ('workflow', 'xforms_session_couch_id', 'reminder_id', 'chat_user_id',
-                  'ignore_opt_out', 'location_id', 'messaging_subevent_id')
+                  'ignore_opt_out', 'location_id', 'messaging_subevent_id', 'custom_metadata')
         for field in fields:
             value = getattr(metadata, field)
             if value is not None:
                 setattr(msg, field, value)
-        msg.save()
```
Comment (emord) [inline corehq/apps/sms/api.py]: why did you remove this save?

File: custom/icds/messaging/custom_content.py (hunk 1 above)
Comment (gcapalbo) [inline custom/icds/messaging/custom_content.py]: yeah, that's cleaner, updated

File: corehq/apps/sms/api.py (hunk 2 above)
Comment (gcapalbo) [inline corehq/apps/sms/api.py]: i noticed this when adding the metadata, it's an unnecessary save as the message is always saved later on after processing, and this prevents one hit to the main db

**Review by emord — APPROVED** [review]
